Normative: Accept only Strings that denote finite values (#146)
tc39/proposal-amount: README.md

* `value: number | bigint | string | ReadonlyArray<number | bigint | string>` —
  The numerical value of the amount.
  The type of the value used in the constructor is retained,
  except that any value that was potentially affected by precision options is a String.

>   A Number `value` can be non-finite (`Infinity`, `-Infinity`, or `NaN`).

Reviewer: Suggestion for better indicating confinement of non-finite values to the Number type:
```suggestion
  except that any value that was potentially affected by precision options is a String.
  A non-finite value is always represented as a Number (`Infinity`, `-Infinity`, or `NaN`).
```

* `value` (Number or BigInt or String): The numerical value of the amount.
  The type of the value used in the constructor is retained,
  except that any value that was potentially affected by precision options is a String.
  Only a Number `value` can be non-finite (`Infinity`, `-Infinity`, or `NaN`);

>   String representations of non-finite values are rejected.

Reviewer: Rejection is _behavior_ and therefore seems inappropriate for section "Properties".
```suggestion
  A Number `value` can be non-finite (`Infinity`, `-Infinity`, or `NaN`).
```

  Attempting to construct an Amount from a `value` that is not a Number or BigInt or String will throw a TypeError.
  When constructing an Amount from a String `value`,
  the string must be exactly a numeric literal denoting a finite value:
  a [StrNumericLiteral](https://tc39.es/ecma262/#prod-StrNumericLiteral)
  other than `"Infinity"`, `"+Infinity"`, or `"-Infinity"`;
  anything else (including `"NaN"`, the empty string,

>   and strings with leading or trailing white space) throws a RangeError.

Reviewer: Grammar nit:
```suggestion
  other than `"Infinity"`, `"+Infinity"`, or `"-Infinity"`.
  Anything else (including `"NaN"`, the empty string,
  and strings with leading or trailing white space) throws a RangeError.
```
